llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1022,6 +1023,8 @@ bool VPWidenIntrinsicRecipe::onlyFirstLaneUsed(const VPValue *Op) const {
   assert(is_contained(operands(), Op) && "Op must be an operand of the recipe");
   // Vector predication intrinsics only demand the the first lane the last
   // operand (the EVL operand).
+  if (VectorIntrinsicID == Intrinsic::experimental_vp_splat)
+    return Op == getOperand(0);
   return VPIntrinsic::isVPIntrinsic(VectorIntrinsicID) &&
          Op == getOperand(getNumOperands() - 1);
```

Comment:
https://github.com/llvm/llvm-project/pull/110489#discussion_r1806215366

Reply:
I think the condition `Op == getOperand(getNumOperands() - 1)` only checks for EVL operand, but for vp.splat, the first operand is also a scalar use.

Reply:
So where you will need this?
IIUC, If this change is only for `VPIntrinsicRecipe::execute`, the change in `isVectorIntrinsicWithScalarOpAtArg` should be good enough. 
```
    if (isVectorIntrinsicWithScalarOpAtArg(VectorIntrinsicID, I.index()))
      Arg = State.get(I.value(), VPLane(0));
```
Is there anything else I'm missing that needs this change?